pages/api/token/temporary.ts

```diff
@@ -18,37 +31,41 @@ import { MiddlewareProps } from "@lib/types";
  * @param email - the email for the user making the request, found in the body of the api request
  * @returns a promise
  */
-const handler = async (req: NextApiRequest, res: NextApiResponse, { formID }: MiddlewareProps) => {
-  try {
-    if (formID == undefined) {
-      return res.status(403).json({ error: "Invalid form request." });
-    }
-    if (!process.env.TOKEN_SECRET) {
-      return res.status(500).json({ error: "Server cannot respond at this time" });
-    }
-
-    const email = req.body["email"];
-    if (!email) {
-      res.status(400).json({ error: "Invalid payload" });
-    }
+const handler = async (
+  req: NextApiRequest,
+  res: NextApiResponse,
+  { formID, email }: MiddlewareProps
+): Promise<void> => {
+  assert(email); // Middleware function `requestContainsEmail` ensures that email is defined
```

Comment:
There are good points on both sides of the argument on whether `assert` should be used in production code.  I noticed however that these `asserts` replaced previous code that verified prop existance and either short circuited, by returning early, and returned the appropriate HTTP response code depending on the Error.
The way the current `assert`s are implemented they are outside of the `try` block and are unhandld in the current function.  There is also no custom message in the `assert`s so when the Assertion Error is thrown it may simply show `Assertion Error: undefined == true`

Reply:
The previous code has just been moved in different places which are then ran in a specific order that insures the variables will be defined. Adding custom messages is definitely a good idea though. I initially wanted to test this approach as the beginning of each function was just a bunch of checks with different exit paths and it was kind of difficult to read through and understand what was happening.
Let me know if you still prefer the old solution?

Reply:
Also the functions in which I used the asserts are local to the file so the impact is limited if we were to change the logical order in
````
export default middleware(
  [
    cors({ allowedMethods: ["POST"] }),
    requestContainsEmail(),
    canLogIn(),
    hasValidFormAccessToken(),
  ],
  handler
);
````